pkg/gds/store/trtl/iterator_test.go

```diff
@@ -0,0 +1,345 @@
+package trtl_test
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/trisacrypto/directory/pkg/gds/store/trtl"
+	"github.com/trisacrypto/directory/pkg/trtl/pb/v1"
+)
+
+func (s *trtlStoreTestSuite) TestBatchIterator() {
+	require := s.Require()
+
+	// Connect to bufconn and get a trtl gRPC client
+	require.NoError(s.grpc.Connect())
+	client := pb.NewTrtlClient(s.grpc.Conn)
+
+	// Iterate over an empty namespace
+	iter := trtl.NewTrtlBatchIterator(client, "empty")
+	require.False(iter.Seek([]byte("foo")), "Seek to non-existent key should return false")
+	require.Nil(iter.Key(), "Key should be nil when iterator is exhausted")
+	require.Nil(iter.Value(), "Value should be nil when iterator is exhausted")
+	require.False(iter.Next(), "Next should return false when iterator is exhausted")
+	require.Nil(iter.Value(), "Value should be nil when iterator is exhausted")
+	require.False(iter.Prev(), "Prev should return false when iterator is exhausted")
+	require.Nil(iter.Value(), "Value should be nil when iterator is exhausted")
```

Comment:
Perhaps the tests on 22, 24, and 26 should return slightly different errors so that we can more easily distinguish what went wrong if something does?

Reply:
Good point, I'll change the wording a bit.